_episodes/23-continuous-integration-automated-testing.md

```diff
@@ -236,4 +236,43 @@ You should talk to your system administrator to setup dedicated runners for you.
 You can also use gitlab.com's runners for free if you provide your credit/debit card details,
 read more [here](https://about.gitlab.com/pricing/#why-do-i-need-to-enter-credit-debit-card-details-for-free-compute-minutes)
 
+> ## Exercise (Optional): Install self-managed GitLab runner.
+> Please refer to [this guide for various choices on how to install a local runner](https://docs.gitlab.com/runner/install/docker.html)
+>  
+> These are the steps that will work for most setups:
+> 
+> ### 0. Make sure you have docker installed on your machine.
+> [TNO Coding guild Docker guide](https://codingguild.tno.nl/coding-at-tno/programming-environments/?h=docker#docker (VPN needed))
+> 
+> ### 1. Install the GitLab runner through docker 
+> Optional: In MacOS systems, /srv does not exist by default. Create /private/srv, or another private directory, for setup.
+> 
+> ```
+> docker run -d --name gitlab-runner --restart always \
+>   -v /srv/gitlab-runner/config:/etc/gitlab-runner \
+>   -v /var/run/docker.sock:/var/run/docker.sock \
+>   gitlab/gitlab-runner:latest
```

Comment:
When I run this locally, I see these messages in the Docker container log, repeated every 3s:
```log
2024-11-16 11:32:23 ERROR: Failed to load config stat /etc/gitlab-runner/config.toml: no such file or directory  builds=0 max_builds=1
```
Not sure this is expected. It might be related to MacOS (see previous comment).

Reply:
I am also not sure. But like I said, I think this is quite advanced stuff and everyone will encounter different errors.

Reply:
I wrote it down as a point to discussin the calibration meeitnbg